apps/api/src/controllers/leaderboard.controller.ts

```diff
@@ -0,0 +1,162 @@
+import { NextFunction, Request, Response } from "express";
+
+import { SelectAgent } from "@/database/schema/core/types.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import { ServiceRegistry } from "@/services/index.js";
+import {
+  AgentMetadata,
+  AgentRank,
+  CompetitionAgentsParamsSchema,
+  LeaderboardParamsSchema,
+} from "@/types/index.js";
+
+export function makeLeaderboardController(services: ServiceRegistry) {
+  /**
+   * Leaderboard Controller
+   * Handles global leaderboard operations.
+   */
+  return {
+    /**
+     * Get global leaderboard across all relevant competitions.
+     * This endpoint is publicly accessible for read-only purposes.
+     * @param req Request object (authentication is optional for this endpoint)
+     * @param res Express response object
+     * @param next Express next function
+     */
+    async getGlobalLeaderboard(
+      req: Request,
+      res: Response,
+      next: NextFunction,
+    ) {
+      try {
+        // Validate query parameters
+        const { success, data, error } = LeaderboardParamsSchema.safeParse(
+          req.query,
+        );
+        if (!success) {
+          throw new ApiError(400, `Invalid request format: ${error.message}`);
+        }
+        const queryParams = data;
+
+        // Get global stats (i.e., generic competition accumulations)
+        const stats = await services.leaderboardService.getGlobalStats(
+          queryParams.type,
+        );
+
+        if (stats.competitionIds.length === 0) {
+          throw new ApiError(404, "No competitions found");
+        }
+
+        // Collect agents and their metrics across all competitions
+        const agentMetricsMap = new Map<string, AgentRank>();
+
+        // Process each competition (need to get all agents in order to properly calculate metrics)
+        await Promise.all(
```

Comment:
not the most efficient logic here, but since it's temporary, i figured it'd be okay to keep w/o optimizing. this is where we copy the per-competition leaderboard/agent metrics logic, but at the end, we use the combined PnL % as the rank.